### ISSUE TITLE
ES6 template strings fail parsing, when there's another parse error

### ISSUE THREAD

**Issue by supersteves** [google/closure-compiler#1136]
I'm using reserved word 'type' and this causes template strings parsing to generate false errors:

``` javascript
function f({type}) {

  var s = `translate(${this._position.x}, ${this._position.y})`);

};
```

```
input0:1: ERROR - Parse error. cannot use keyword 'type' here.
function f({type}) {
                ^

input0:3: ERROR - Parse error. primary expression expected
  var s = `translate(${this._position.x}, ${this._position.y})`);
                                         ^

input0:3: ERROR - Parse error. Semi-colon expected
  var s = `translate(${this._position.x}, ${this._position.y})`);
                                          ^

input0:3: ERROR - Parse error. Semi-colon expected
  var s = `translate(${this._position.x}, ${this._position.y})`);
                                           ^

input0:3: ERROR - Parse error. primary expression expected
  var s = `translate(${this._position.x}, ${this._position.y})`);
                                                              ^

input0:3: ERROR - Parse error. Unterminated template literal
  var s = `translate(${this._position.x}, ${this._position.y})`);
                                                               ^
```

https://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520f(%257Btype%257D)%2520%257B%250A%250A%2520%2520var%2520s%2520%253D%2520%2560translate(%2524%257Bthis._position.x%257D%252C%2520%2524%257Bthis._position.y%257D)%2560)%253B%250A%250A%257D%253B%250A%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26IDE_MODE%3D1%26LANG_IN_IS_ES6%3D1

If I turn off IDE mode, the 5 template string parse errors disappear when running locally, or crash using the debugger:

https://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520f(%257Btype%257D)%2520%257B%250A%250A%2520%2520var%2520s%2520%253D%2520%2560translate(%2524%257Bthis._position.x%257D%252C%2520%2524%257Bthis._position.y%257D)%2560)%253B%250A%250A%257D%253B%250A%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26LANG_IN_IS_ES6%3D1

```
java.lang.IllegalArgumentException: Unexpected control flow graph root BLOCK [synthetic: 1]
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:150)
	at com.google.javascript.jscomp.ControlFlowAnalysis.process(ControlFlowAnalysis.java:148)
	at com.google.javascript.jscomp.Compiler.getAstDotGraph(Compiler.java:2325)
	at com.google.javascript.jscomp.debugger.CompilationServlet.service(CompilationServlet.java:101)
```

If I fix the reserved word, it works fine:

https://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520f(%257Bthetype%257D)%2520%257B%250A%250A%2520%2520var%2520s%2520%253D%2520%2560translate(%2524%257Bthis._position.x%257D%252C%2520%2524%257Bthis._position.y%257D)%2560%253B%250A%250A%257D%253B%250A%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26IDE_MODE%3D1%26LANG_IN_IS_ES6%3D1

**Comment by Dominator008** [comment]
"type" is not a reserved word but a TypeScript specific keyword. I'm fixing this via #1137.

**Comment by supersteves** [comment]
Understood. That was fast! Thanks.

**Comment by MatrixFrog** [comment]
So this should be fixed now, I think?

**Comment by supersteves** [comment]
I think so.  Well, at least, no longer a problem with this type keyword (confirmed) fix.

(I thought it was a bit weird that the keyword "type" issue would cause ES6 template string parsing to also fail, but I guess the parser is complex)
